dvc/repo/plots/__init__.py

```diff
@@ -95,16 +108,28 @@ def render(data, revs=None, props=None, templates=None):
 
         return result
 
-    def show(self, targets=None, revs=None, props=None, templates=None):
+    def show(
+        self,
+        targets=None,
+        revs=None,
+        props=None,
+        templates=None,
+        recursive=False,
+    ):
         from .data import NoMetricInHistoryError
 
-        data = self.collect(targets, revs)
+        data = self.collect(targets, revs, recursive)
 
         # If any mentioned plot doesn't have any data then that's an error
         targets = [targets] if isinstance(targets, str) else targets or []
         for target in targets:
             rpath = relpath(target, self.repo.root_dir)
-            if not any("data" in d[rpath] for d in data.values()):
+            if not any(
+                "data" in rev_data[key]
+                for rev_data in data.values()
+                for key, d in rev_data.items()
+                if rpath in key
+            ):
```

Comment:
since we support `dir plots` we need to consider possibility that data paths overlap.
Eg:
`dvc plots show directory` might result in {"directory/plot1.json":{...}, "directory/plot2.json":{}}.